### PR TITLE
[openseadragon] update types for v.3.0

### PR DESCRIPTION
**Pull request by alvaromartmart**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/openseadragon/openseadragon/issues/1985
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
> TODO: v.3.0 has not yet been released

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by iangilman** [comment]
Looking good to me so far! @msalsbery can you please take a look, as you are more familiar with the event handler changes, which are the bulk of this patch?

**Comment by msalsbery** [comment]
> Looking good to me so far! @msalsbery can you please take a look, as you are more familiar with the event handler changes, which are the bulk of this patch?

Will do, thanks for the heads-up!

File: types/openseadragon/index.d.ts [hunk 1]
```diff
@@ -603,30 +649,52 @@ declare namespace OpenSeadragon {
...
         keyHandler: (event: OSDEvent<any>) => void;
         keyUpHandler: (event: OSDEvent<any>) => void;
         moveHandler: (event: OSDEvent<any>) => void;
         nonPrimaryPressHandler: (event: OSDEvent<any>) => void;
         nonPrimaryReleaseHandler: (event: OSDEvent<any>) => void;
+        overHandler: (event: OSDEvent<any>) => void;
+        outHandler: (event: OSDEvent<any>) => void;
         pinchHandler: (event: OSDEvent<any>) => void;
         pressHandler: (event: OSDEvent<any>) => void;
+        preProcessEventHandler: (event: OSDEvent<any>) => void;
```
Comment (msalsbery) [inline types/openseadragon/index.d.ts:673]: preProcessEventHandler will always get a EventProcessInfo object

**Review by msalsbery — COMMENTED** [review]
@alvaromartmart

I made one comment in the code changes - not a show stopper for sure ;-)  Looks good to me, thank you!

File: types/openseadragon/index.d.ts (hunk 1 above)
Comment (alvaromartmart) [reply types/openseadragon/index.d.ts]: > preProcessEventHandler will always get a EventProcessInfo object

Thank you @msalsbery, I think it's fixed now

**Comment by claycoleman** [comment]
> // Type definitions for OpenSeadragon 2.4

the top of the file still has the old version number, btw!

**Comment by claycoleman** [comment]
@alvaromartmart mind updating and publishing this PR, as openseadragon/openseadragon#1985 has now been merged and 3.0.0 has been released?

**Comment by alvaromartmart** [comment]
> @alvaromartmart mind updating and publishing this PR, as [openseadragon/openseadragon#1985](https://github.com/openseadragon/openseadragon/issues/1985) has now been merged and 3.0.0 has been released?

Sure, I plan to review this once I have some spare time!

**Comment by peterblazejewicz** [comment]
Please amend the version, so it's properly resolved by tooling

**Comment by alvaromartmart** [comment]
Ok, I had the chance to review it a bit, I think it looks pretty good.

To be honest I don't have that much time now to perform more extensive testing, and I no longer use OpenSeadragon on a daily basis. It would be **very** helpful if someone could try to use this version of the types against some advanced openseadragon scenarios, specially those using new features / main changes introduced by v.3.0. @msalsbery / @iangilman  maybe you could help me with that?

**Review by iangilman — COMMENTED** [review]
Everything looks good from what I can tell. I'm familiar with OpenSeadragon, but I don't use it with TypeScript, so I might not be a definitive source here.

@alvaromartmart thank you for putting this together!

**Review by peterblazejewicz — APPROVED** [review]

**Comment by alvaromartmart** [comment]
Ready to merge

**Pull request merged**